clippy_lints/src/misc.rs

```diff
@@ -519,7 +519,8 @@ fn is_signum(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
 }
 
 fn is_float(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
-    matches!(walk_ptrs_ty(cx.tables.expr_ty(expr)).sty, ty::Float(_))
+    let value = &walk_ptrs_ty(cx.tables.expr_ty(expr)).sty;
+    matches!(value, ty::Array(_ty, _)) || matches!(value, ty::Float(_))
```

Comment:
Hey, @flip1995. All done with the update. This is what I have now.

Reply:
You also have to match `_ty.node` to `Float(_)`:
```suggestion
    matches!(value, ty::Array(ty, _)) && matches!(ty.node, ty::Float(_)) || matches!(value, ty::Float(_))
```

Reply:
You may also want to `walk_ptrs_ty(ty)`, since an `Array` of `&Float`s shouldn't be compared either.

Reply:
I think I understand what I'm supposed to do. 

`matches!(value, ty::Array(ty, _))` matches an array

`matches!(ty.node, ty::Float(_))` is to match when the value inside the array is a Float. 

<hr>

The issue I'm having right now is getting the second part, checking if it is a Float. 

When I match against `ty.node`, I get this error: _expected value, found module **ty**_

This is what I get when print the value of `walk_ptrs_ty`, when it's an array. 
 
```Rust
Array(u8, Const { ty: usize, val: Scalar(0x0000000000000004) })
```

Reply:
You can ignore the `Const` struct, since it is only the length of the array. The Rust type system already forbids the comparison of two arrays with different lengths, so we don't have to take care about that.

You have to match the `u8`. Note, that the `u8` is only displayed because of the pretty printing of types. In fact it is just another `ty::Ty`.

On the error message: You get this, because you shadow the import of `ty` with the variable name `ty`. Renaming the variable `ty` with `arr_ty` or similar should solve this.

Reply:
Thanks. I will get to work on it.